WordPress/src/main/java/org/wordpress/android/util/experiments/BiasAAExperiment.kt

```diff
@@ -0,0 +1,18 @@
+package org.wordpress.android.util.experiments
+
+import org.wordpress.android.fluxc.model.experiments.Variation.Treatment
+import javax.inject.Inject
+
+class BiasAAExperiment
+@Inject constructor(
+    exPlat: ExPlat
+) : Experiment(
+        id = 20085,
+        name = "explat_test_aa_weekly_wpandroid_2021_week_06",
+        exPlat
+) {
+    fun getEventProperties() = mapOf(
+            "experiment_id" to id,
+            "experiment_variation" to getVariation().let { if (it is Treatment) it.name else "control" }
```

Comment:
For most experiments, where the only variations are `Control` and `Treatment`, I see no need to define them explicitly, as they can be easily handled like this:

```kotlin
when (myExperiment.getVariation()) {
    Control -> handleControl()    
    else -> handleTreatment()
}
```

For experiments with multiple variations, it would still be pretty simple:

```kotlin
when (myExperiment.getVariation()) {
    Treatment("treatment_one") -> handleTreatmentOne()
    Treatment("treatment_two") -> handleTreatmentTwo()
    else -> handleControl()
}
```

But we could also introduce an enum on top of the regular variation and achieve a similar result to what you suggested:

```kotlin
// Experiment definition
class MyExperiment(exPlat: ExPlat) : Experiment("my_experiment", exPlat) {
    fun getVariation(): EnumVariation {
        return when (super.getVariation(false)) {
            Treatment("treatment_one") -> TREATMENT_ONE
            Treatment("treatment_two") -> TREATMENT_TWO
            else -> CONTROL
        }
    }

    enum class EnumVariation {
        CONTROL, TREATMENT_ONE, TREATMENT_TWO
    }
}

// How we would handle it
when (myExperiment.getVariation()) {
    TREATMENT_ONE -> handleTreatmentOne()
    TREATMENT_TWO -> handleTreatmentTwo()
    else -> handleControl()
}
```

Keeping this flexible and having these options seems like a good deal for me. Let me know what you think.

Reply:
I think all of these calls are ok if they happen within the Experiment. However, I think calls like 
```
when (myExperiment.getVariation()) {
    Treatment("treatment_one") -> handleTreatmentOne()
    Treatment("treatment_two") -> handleTreatmentTwo()
    else -> handleControl()
}
```
it is simple but quite error prone. Especially since we'll have multiple places calling these. For experiments with just one treatment I think it would be much neater to just expose one method `isTreatment` instead of exposing `getVariations`. The same is for more treatments. Either having an enum or exposing methods like `isTreatmentA`, `isTreatmentB` and so on. Enums are not mandatory but I'd really consider making the `getVariations` method private so any future developers have to think about what they are exposing to make sure there is any unintentional mistake. Sometimes less flexibility is good for more safety :-).

Reply:
>I'd really consider making the getVariations method private so any future developers have to think about what they are exposing to make sure there is any unintentional mistake.

That's a solid point.

My intention was for us to have a standard, default way of handling variations that would work for most cases, where we wouldn't need to call `getVariations` from multiple places and wouldn't have more than two variations (this feature is not even available at the moment). But I guess making `getVariations` private and introducing a basic `isControl` method could probably achieve the same result I was looking for.

Another option I had in mind would be to introduce a `ABExperiment` class that extends `Experiment` and implements those methods by default. Then we could also introduce a `MultivariateExperiment` later on, that could enforce safety by using enums or something similar.